Fix:

warning: cast to smaller integer type 'char' from 'void *' [-Wvoid-pointer-to-int-cast]
error: incompatible integer to pointer conversion assigning to 'char *' from 'char' [-Wint-conversion]
--- fdutil.c.orig	2005-07-28 12:29:07.000000000 -0500
+++ fdutil.c	2024-10-02 01:29:05.000000000 -0500
@@ -84,7 +84,7 @@
 			buf[i] = c;
 		}
 	}
-	buf[i] = (char)NULL;
+	buf[i] = '\0';
 
 	return buf;
 }
--- fileconfig.c.orig	2005-07-28 12:29:07.000000000 -0500
+++ fileconfig.c	2024-10-02 01:30:28.000000000 -0500
@@ -168,7 +168,7 @@
 		}
 		else if ((tempo = strchr(buf,'#')))
 		{
-			*tempo = (char)NULL;
+			*tempo = '\0';
 		}
 		else if(strchr(buf,'=') == (char*)NULL)
 		{
--- main.c.orig	2005-07-28 12:29:07.000000000 -0500
+++ main.c	2024-10-02 01:19:55.000000000 -0500
@@ -52,11 +52,11 @@
 
 	str_init(&msg_buffer, MAX_MSG_LEN);
 
-	serverinfo.host = (char) NULL;
-	serverinfo.fromaddr = (char) NULL;
-	serverinfo.domain = (char) NULL;
-	serverinfo.auth_user = (char) NULL;
-	serverinfo.auth_pass = (char) NULL;
+	serverinfo.host = NULL;
+	serverinfo.fromaddr = NULL;
+	serverinfo.domain = NULL;
+	serverinfo.auth_user = NULL;
+	serverinfo.auth_pass = NULL;
 	serverinfo.auth_mech = SASL_DEFAULT;
 	serverinfo.port = 0;
 	serverinfo.num_rcpts = 0;
--- original.c.orig	2005-07-28 12:29:07.000000000 -0500
+++ original.c	2024-10-02 01:19:51.000000000 -0500
@@ -461,8 +461,8 @@
 			else
 			{
 				/* HELO worked, so we NULL all authentication vars and warn to syslog */
-				serverinfo->auth_user = (char)NULL;
-				serverinfo->auth_pass = (char)NULL;
+				serverinfo->auth_user = NULL;
+				serverinfo->auth_pass = NULL;
 
 				log_msg(LOG_WARNING,"nbSMTP will continue without authentication");
 
--- util.c.orig	2005-07-28 12:29:07.000000000 -0500
+++ util.c	2024-10-02 01:28:13.000000000 -0500
@@ -180,7 +180,7 @@
 				*p='X';
 			}
 
-			*p = (char)NULL;
+			*p = '\0';
 		}
 	}
 
